src/plugins/intel_cpu/src/nodes/executors/shl/shl.hpp

```diff
@@ -163,6 +163,47 @@ struct ShlTensor : public ShlStructure<csinn_tensor*> {
         OPENVINO_ASSERT(get()->dim_count < MAX_DIM, "Shl supports shapes with rank less or equal to 8");
         for (int i = 0; i < get()->dim_count; ++i)
             get()->dim[i] = static_cast<int32_t>(shape[i]);
+    };
+};
+
+// virtual base class for different kinds of params
+struct IShlParams {
+public:
+    virtual ~IShlParams() = default;
+    virtual void reset(void* t) = 0;
+    virtual void* get(bool allow_empty = false) const = 0;
+    virtual void setAPI(csinn_api_enum api) = 0;
```

Comment:
It should be in private section

Reply:
Maybe only `reset` and `setAPI` should be in private section? I think `IShlParams()` and `get()` should be in public section.

Reply:
I'd say that this interface class shouldn't have `reset` method because we don't need to call this method in executors or somewhere except private class methods. Then I think we can remove `reset` from `IShlParams`.
As for `setAPI` - now it's private method in interface class. So sounds like it's not needed method in `IShlParams`.
We can move this method to  private section `ShlParams` class and remove from `IShlParams`. 

Thus, the interface class `IShlParams` will have only on needed common method - `get()`.
What do you think?

Reply:
I think this way is much clearer, I will update soon!